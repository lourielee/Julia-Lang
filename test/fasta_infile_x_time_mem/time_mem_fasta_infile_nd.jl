include("/Users/lauriecannon/projects/Julia/Julia_binf/src/fasta_infile_no_dependency.jl")

outfile = open("test/fasta_infile_x_time_mem/outfiles/time_mem_fasta_infile_nd.out", "w")

println(outfile,"initialize():")
println(outfile, @elapsed initialize("data/SARS-CoV-2/protein/covid_proteins.fasta"))
println(outfile,"get_ID:")
println(outfile, @elapsed get_ID("data/SARS-CoV-2/protein/covid_proteins.fasta"))
println(outfile, "get_SEQ():")
println(outfile, @elapsed get_SEQ("data/SARS-CoV-2/protein/covid_proteins.fasta"))

#measure multiple invocations because compilation time may affect first @time
println(outfile, "get_ID_SEQ():")
println(outfile, @elapsed get_ID_DESC("data/SARS-CoV-2/protein/covid_proteins.fasta"))
println(outfile, @elapsed get_ID_DESC("data/SARS-CoV-2/protein/covid_proteins.fasta"))
println(outfile, @elapsed get_ID_DESC("data/SARS-CoV-2/protein/covid_proteins.fasta"))
println(outfile, @elapsed get_ID_DESC("data/SARS-CoV-2/protein/covid_proteins.fasta"))
close(outfile)
