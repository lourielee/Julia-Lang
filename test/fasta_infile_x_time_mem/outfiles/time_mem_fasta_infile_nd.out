initialize():
0.00028625
get_ID:
0.000251112
get_SEQ():
0.000267929
get_ID_SEQ():
0.000414563
0.000407137
0.000406895
0.00042399
